GitCommands/RevisionReader.cs

```diff
@@ -327,11 +325,11 @@ int CountParents(ref ReadOnlySpan<byte> array, int baseOffset)
             #region Timestamps
 
             // Lines 2 and 3 are timestamps, as decimal ASCII seconds since the unix epoch, each terminated by `\n`
-            var authorUnixTime = ParseUnixDateTime(ref array);
-            var commitUnixTime = ParseUnixDateTime(ref array);
+            var authorUnixTime = ParseUnixDateTime(array);
+            var commitUnixTime = ParseUnixDateTime(array);
 
             [MethodImpl(MethodImplOptions.AggressiveInlining)]
-            long ParseUnixDateTime(ref ReadOnlySpan<byte> array)
+            long ParseUnixDateTime(in ReadOnlySpan<byte> array)
```

Comment:
ROS is a struct, so it has to be be `ref`, otherwise it will be copied

Reply:
ROS is a `readonly ref struct` so I interpret it as in will use a reference.
https://docs.microsoft.com/en-us/dotnet/csharp/write-safe-efficient-code#avoid-defensive-copies
`in` should be specified at the call sites to specify the intent. 

So the ObjectId.cs changes does not have any effect.

Reply:
`ref structs` aren't copied, they are passed by reference

Reply:
https://docs.microsoft.com/dotnet/csharp/write-safe-efficient-code#use-ref-struct-types

Reply:
I interpret the section above that described defensive copies to that `in` should do the same as `ref` here.
If you disagree, I can change, ref will work (but it is not the intent).
If that is changed, I would like to change StringLineReader internal class to local methods.

Reply:
> `ref structs` aren't copied, they are passed by reference

Ref structs can be copied on the stack. They just cannot escape the stack, which gives them guarantees around their lifetimes.

Both ref and in can avoid the copy for a readonly struct (ref or not). I suspect in to be the better choice here.

Reply:
I stand corrected :)